tests/unit/restapi/v1/test_queue.py

```diff
@@ -0,0 +1,516 @@
+      default group.
+    - The returned list of queues matches the expected list owned by the default group.
+    """
+    queue1_expected = registered_queues["queue1"].get_json()
+    queue2_expected = registered_queues["queue2"].get_json()
+    queue3_expected = registered_queues["queue3"].get_json()
+    queue_expected_list = [queue1_expected, queue2_expected, queue3_expected]
+
+    search_parameters = {"group_id": auth_account["default_group_id"]}
+
+    assert_retrieving_all_queues_works(
+        client, expected=queue_expected_list, G=search_parameters
+    )
+
+
+@pytest.mark.v1
+def test_queue_id_get(
+    client: FlaskClient,
+    db: SQLAlchemy,
+    auth_account: dict[str, Any],
+    registered_queues: dict[str, Any],
+) -> None:
+    """Test that queues can be retrieved by their unique ID.
+
+    Given an authenticated user and registered queues, this test validates the following
+        sequence of actions:
+
+    - The user is able to retrieve single queue by its ID.
+    - The response is a single queue with a matching ID.
+    """
+    queue3_expected = registered_queues["queue3"].get_json()
+
+    assert_retrieving_queue_by_id_works(
+        client, queue_id=queue3_expected["queueId"], expected=queue3_expected
+    )
+
+
+@pytest.mark.v1
+def test_cannot_register_existing_queue_name(
+    client: FlaskClient,
+    db: SQLAlchemy,
+    auth_account: dict[str, Any],
+    registered_queues: dict[str, Any],
+) -> None:
+    """Test that registering a queue with an existing name fails.
+
+    Given an authenticated user and registered queues, this test validates the following
+        sequence of actions:
+
+    - The user attempts to register a second queue with the same name.
+    - The request fails with an appropriate error message and response code.
+    """
+    queue1_expected = registered_queues["queue1"].get_json()
+
+    assert_registering_existing_queue_name_fails(client, name=queue1_expected["name"])
+
+
+@pytest.mark.v1
+def test_rename_queue(
+    client: FlaskClient,
+    db: SQLAlchemy,
+    auth_account: dict[str, Any],
+    registered_queues: dict[str, Any],
+) -> None:
+    """Test that a queue can be renamed.
+
+    Given an authenticated user and registered queues, this test validates the following
+        sequence of actions:
+
+    - The user issues a request to change the name of a queue.
+    - The user retrieves information about the same queue and it reflects the name
+      change.
+    - The user issues a request to change the name of a queue to an existing queue's
+      name.
+    - The request fails with an appropriate error message and response code.
+    """
+    updated_queue_name = "tensorflow_gpu"
+    queue_to_rename = registered_queues["queue1"].json()
+    existing_queue = registered_queues["queue2"].json()
+
+    rename_queue(
+        client, queue_id=queue_to_rename["queueId"], new_name=updated_queue_name
+    )
+
```

Comment:
Minor nitpick, the whitespace between these functions can be removed. (comment applies to other test functions as well).